src/components/[guild]/RoleCard/components/Reward.tsx

```diff
@@ -260,7 +261,11 @@ const RewardWrapper = ({ platform, ...props }: RewardProps) => {
   const Component =
     rewards[PlatformType[guildPlatform?.platformId]].RoleCardComponent ?? Reward
 
-  return <Component platform={platformWithGuildPlatform} {...props} />
+  return (
+    <ApiRequirementHandlerProvider roleId={platformWithGuildPlatform.roleId}>
```

Comment:
If I understand correctly this is needed because of the DynamicTag changes, so it works when rendered on the RoleCards too. But this way it's included for every end user too which could negatively affect bundle size & performance, can we use a condition in DynamicReward or a prop based solution instead?

Reply:
The problem is that besides the create handler, it also provides the list of requirements itself as well, which is not admin-only.
We can, however, wrap only the `DynamicTag` with the provider, so it would only affect dynamic rewards.